src/shared/components/species-name/SpeciesName.tsx

```diff
@@ -0,0 +1,160 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import classNames from 'classnames';
+import type { HTMLAttributes } from 'react';
+
+import { useAppSelector } from 'src/store';
+
+import { getSpeciesNameDisplayOption } from 'src/content/app/species-selector/state/species-selector-general-slice/speciesSelectorGeneralSelectors';
+
+import {
+  AssemblyAccessionId,
+  CommonName,
+  ScientificName,
+  SpeciesType,
+  AssemblyName,
+  SpeciesReference
+} from '../species-name-parts';
+
+import type { CommittedItem } from 'src/content/app/species-selector/types/committedItem';
+import type { SpeciesNameDisplayOption } from 'src/content/app/species-selector/types/speciesNameDisplayOption';
+
+import styles from './SpeciesName.module.css';
+
+export type Props = HTMLAttributes<HTMLSpanElement> & {
+  species: Pick<
+    CommittedItem,
+    'common_name' | 'scientific_name' | 'assembly' | 'type' | 'is_reference'
+  >;
+  speciesNameDisplayOption: SpeciesNameDisplayOption;
+};
+
+// Exported component that is not connected to redux. For convenience while testing.
+export const SpeciesName = (props: Props) => {
+  const {
+    species,
+    speciesNameDisplayOption,
+    className: classNameFromProps,
+    ...otherProps
+  } = props;
+
+  const componentClasses = classNames(styles.speciesName, classNameFromProps);
+
+  return (
+    <span className={componentClasses} {...otherProps}>
+      <Content species={species} displayOption={speciesNameDisplayOption} />
+    </span>
+  );
+};
+
+const Content = (props: {
+  species: Props['species'];
+  displayOption: SpeciesNameDisplayOption;
+}) => {
+  const { species, displayOption } = props;
+  const scientificNameClasses = classNames(styles.prominent, styles.italic);
+
+  const scientificNameElement = (
+    <ScientificName
+      scientific_name={species.scientific_name}
+      className={scientificNameClasses}
+    />
+  );
+
+  if (displayOption === 'common-name_assembly-name') {
+    return (
+      <>
+        <CommonName
+          common_name={species.common_name}
+          fallback={scientificNameElement}
+          className={styles.prominent}
+        />
+        <AssemblyName assembly={species.assembly} className={styles.assembly} />
+      </>
+    );
+  } else if (displayOption === 'common-name_type_assembly-name') {
+    return (
+      <>
+        <CommonName
+          common_name={species.common_name}
+          fallback={scientificNameElement}
+          className={styles.prominent}
+        />
+        {!!(species.type || species.is_reference) && (
```

Comment:
Does this mean that `SpeciesType` or `SpeciesReference` can be rendered even if they are not available?

Reply:
Each of them should return `null` if it isn't available.